pkg/api/componentreadiness/triage.go

```diff
@@ -0,0 +1,174 @@
+}
+
+func ListTriages(dbc *db.DB) ([]models.Triage, error) {
+	triages, err := query.ListTriages(dbc)
+	for i := range triages {
+		injectHATEOASLinks(&triages[i])
+	}
+	return triages, err
+}
+
+// validateTriage ensures the Triage record coming into the API appears valid. Small
+// changes in logic for create vs update are controlled by the update param.
+func validateTriage(triage models.Triage, update bool) error {
+	create := !update
+	if create && triage.ID > 0 {
+		return fmt.Errorf("cannot specify an id for a new triage record, one will be autogenerated")
+	}
+	if update && triage.ID == 0 {
+		return fmt.Errorf("must specify an id for a triage record update")
+	}
+	if triage.URL == "" {
+		return fmt.Errorf("url is required for a triage record")
+	}
+	if !models.ValidTriageType(triage.Type) {
+		return fmt.Errorf("invalid triage type: %s", triage.Type)
+	}
+
+	return nil
+}
+
+func CreateTriage(dbc *db.DB, triage models.Triage) (models.Triage, error) {
+	err := validateTriage(triage, false)
+	if err != nil {
+		log.WithError(err).Error("error validating triage record")
+		return triage, err
+	}
+	// zero out the timestamps, you shouldn't be specifying them, but it's not worth erroring a request over
+	triage.CreatedAt = time.Time{}
+	triage.UpdatedAt = time.Time{}
+
+	// We support linking to regressions by just setting the ID in the request, lookup
+	// full regressions for association.
+	regressionIDs := []uint{}
+	for _, trIDR := range triage.Regressions {
+		regressionIDs = append(regressionIDs, trIDR.ID)
+	}
+	var linkedRegressions []models.TestRegression
+	res := dbc.DB.Where("id IN ?", regressionIDs).Find(&linkedRegressions)
+	if res.Error != nil {
+		log.WithError(res.Error).Errorf("error looking up regression IDs: %v", regressionIDs)
+		return triage, res.Error
+	}
+
+	if len(linkedRegressions) != len(regressionIDs) {
+		err := fmt.Errorf("some of the requested regression IDs were not found: %v",
+			regressionIDs)
+		log.WithError(err).Error("error looking up test regressions during create")
+		return triage, err
+	}
+	triage.Regressions = linkedRegressions
+
+	// If we have a bug in the db matching the url we were given, link them up now.
+	// If not, this should be handled later during the next fetchdata cron job.
+	var bug models.Bug
+	res = dbc.DB.Where("url = ?", triage.URL).First(&bug)
+	switch {
+	case res.Error != nil && !errors.Is(res.Error, gorm.ErrRecordNotFound):
+		log.WithError(res.Error).Errorf("unexpected error looking up bug: %s", triage.URL)
+		return triage, res.Error
+	case res.Error == nil:
+		triage.Bug = &bug
+		triage.BugID = &bug.ID
+	}
+
+	res = dbc.DB.Create(&triage)
+	if res.Error != nil {
+		log.WithError(res.Error).Error("error creating triage record")
+		return triage, res.Error
+	}
+	log.WithField("triageID", triage.ID).Info("triage record created")
+
+	injectHATEOASLinks(&triage)
+	return triage, nil
+}
+
+func UpdateTriage(dbc *db.DB, triage models.Triage) (models.Triage, error) {
```

Comment:
What is the plan for audit create / update?  Maybe we want to get this in and build off of it but do we intend for the audit tracking to be present before enabling?

Reply:
Definitely by the time we roll out for others. I do not have a design in mind yet, something attached to the triage record that covers user and what they did though, including addition/removal of job runs, all stored in postgres.

Reply:
Could I suggest that we work out a minimal design that we use to record who (for the script we just us the service account email id currently) and when?  At least that way we have history of updates and then we can build out specific update types.  Doesn't have to be this PR but feels like it should be a priority before switching over.

Reply:
Yes I'll add a jira and get thinking on this asap.

Reply:
https://issues.redhat.com/browse/TRT-2075